Build fails on macos Catalina due to resgen segfault

**Describe the bug**
After an update on MacOS to Catalina the build process fails. It seems most of the executables are segfaulting as well 

**To Reproduce**
Steps to reproduce the behavior:
Prereq: Macbook pro 2019 
1. clone git repository master
2. call ./build.sh
3. wait until crash 

**Expected behavior**
resgen should not be segfaulting

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Desktop (please complete the following information):**
 - OS: MacOS Catalina 10.15
 - GPU:Radeon Pro Vega 20 4 GB - Intel UHD Graphics 630 1536 MB
 - Backend: [OpenGL/Vulkan]

**Additional context**
Add any other context about the problem here.

The same problem for me. My call stack is attached here, when running `ninja`
<details>
<summary>Output</summary>
[649/1167] Generating matdbg_resources.bin, matdbg...es.S, matdbg_resources.apple.S, matdbg_resources.h
FAILED: libs/matdbg/matdbg_resources.bin libs/matdbg/matdbg_resources.S libs/matdbg/matdbg_resources.apple.S libs/matdbg/matdbg_resources.h
cd /Users/elvischen/filament/out/cmake-release/libs/matdbg && /Users/elvischen/filament/out/cmake-release/tools/resgen/resgen -t -x /Users/elvischen/filament/out/cmake-release/libs/matdbg -p matdbg_resources /Users/elvischen/filament/libs/matdbg/web/style.css /Users/elvischen/filament/libs/matdbg/web/script.js /Users/elvischen/filament/libs/matdbg/web/index.html
/bin/sh: line 1: 88510 Segmentation fault: 11  /Users/elvischen/filament/out/cmake-release/tools/resgen/resgen -t -x /Users/elvischen/filament/out/cmake-release/libs/matdbg -p matdbg_resources /Users/elvischen/filament/libs/matdbg/web/style.css /Users/elvischen/filament/libs/matdbg/web/script.js /Users/elvischen/filament/libs/matdbg/web/index.html
[662/1167] Building CXX object third_party/libassi...p.dir/__/code/Importer/StepFile/StepFileGen1.cpp.o
ninja: build stopped: subcommand failed.
</details>

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -224,6 +224,12 @@ if (NOT CLANG_CL AND NOT MSVC AND NOT WEBGL)
 endif()
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${EXTRA_SANITIZE_OPTIONS}")
 
+# Disable the stack check for macOS to workaround a known issue in clang 11.0.0.
+# See: https://forums.developer.apple.com/thread/121887
+if (APPLE)
+    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}  -fno-stack-check")
+endif()
+
 # ==================================================================================================
 # Linker flags
 # ==================================================================================================

